internal/transactionsubmission/monitor/monitor_service.go

```diff
@@ -0,0 +1,123 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stellar/go/support/log"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/monitor"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/store"
+)
+
+type MonitorService struct {
+	GitCommitHash string
+	Version       string
+	MonitorClient monitor.MonitorServiceInterface
```

Comment:
Calling it a client is a  bit inaccurate. Maybe you should be including this interface?: `monitor.MonitorClient`?

https://github.com/stellar/stellar-disbursement-platform-backend/blob/dc974cf6c9104996e846a615e99a5f396704dfc1/internal/monitor/monitor_service.go#L20-L22

That'd require you to receive the metrics options instead of the service, though. Something similar to this:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/dc974cf6c9104996e846a615e99a5f396704dfc1/internal/monitor/monitor_service.go#L24-L37

Reply:
Or you can use inheritance, in case there's your initial idea:

```suggestion
	monitor.MonitorServiceInterface
```

Inheritance is not very encouraged by some authors, but could be an option here, considering you won't be overriding many functions.